timemachine/fe/single_topology.py

```diff
@@ -0,0 +1,954 @@
+
+    mol: Chem.Mol
+        RDKit molecule
+
+    Returns
+    -------
+    set
+        Set of chiral bonds.
+
+    """
+    # a chiral bond is defined as a bond that
+    # 1) has a single proper torsion term with k > 10 kJ/mol, period=2, phase=3.1415
+    # 2) is not part of a ring system.
+    # the reason why 2) is present is because the planar torsions spanning a
+    # ring system are not used to enforce chiralchemistry, since if we simply
+    # disabled them, we would *still* get the correct chiralchemistry due to steric effects.
+    # consider a benzene devoid of any torsions (proper or improper), or non-bonded
+    # terms, and only angles and bonds are present. The hydrogens would still be correctly placed.
+
+    canonical_ring_bonds = set()
+    for ij in ring_bonds:
+        canonical_ring_bonds.add(dummy.canonicalize_bond(ij))
+
+    planar_bonds_kv = identify_bonds_spanned_by_planar_torsions(proper_idxs, proper_params)
+    canonical_chiral_bonds = set()
+    for k in planar_bonds_kv.keys():
+        if k not in canonical_ring_bonds:
+            src, dst = k
+            src = mol.GetAtomWithIdx(src)
+            dst = mol.GetAtomWithIdx(dst)
+            if src.GetDegree() == 4 or dst.GetDegree() == 4:
+                # if either atom is tetrahedral then it has no pi bonds to conjugate over
+                # exception: weird sulfur groups
+                continue
+            else:
+                canonical_chiral_bonds.add(k)
+
+    return canonical_chiral_bonds
+
+
+def find_chiral_atoms(mol):
+    """
+    Find chiral atoms in a molecule. Chirality is perceived locally, without taking
+    graph-symmetry into account. Eg. even something like CH4 would be considered
+    to be chiral since the hydrogens are distinguishable.
+
+    Returns
+    -------
+    set
+        Set of chiral atoms.
+
+    """
+    mol_geom = geometry.classify_geometry(mol)
+    chiral_atoms = set()
+    for a in mol.GetAtoms():
+        a_idx = a.GetIdx()
+        nbs = a.GetNeighbors()
+        if len(nbs) == 4:
+            chiral_atoms.add(a_idx)
+        elif len(nbs) == 3 and mol_geom[a_idx] == LocalGeometry.G3_PYRAMIDAL:
+            # if in ring, or is sulfur or phosphorus, this may not be guaranteed by
+            if a.GetAtomicNum() == 16 or a.GetAtomicNum() == 15:
+                assert 0, "Pyramidal Phosphorous or Sulfur detected."
+        elif len(nbs) > 4:
+            assert 0, "More than four neighbors found"
+    return chiral_atoms
+
+
+def check_bond_stability(j, k, bond_idxs, bond_params):
+    # tbd this should really be based on the pdf, we should expect to see
+    # near zero probability at bond lengths < 0.5 Angstroms
+    canonical_bonds = dict()
+    for idxs, params in zip(bond_idxs, bond_params):
+        canonical_bonds[dummy.canonicalize_bond(idxs)] = params
+
+    jk = dummy.canonicalize_bond((j, k))
+
+    if jk not in canonical_bonds:
+        return False
+    elif canonical_bonds[jk][0] < 50.0:
```

Comment:
could we add a comment on where `50.0` comes from? (the 0.05 nm is mentioned in the comment above)

Reply:
added comment for both check_bond and  check_angle